### DISCUSSION TITLE
How to highlight a link only if it contains specific content?

### DISCUSSION THREAD

**Discussion by DannySeidel** [typst/typst#4436] [Q&A] [answered]
I'm trying to highlight links in my template only if the link is not an acronym, as the acronyms are linked to their definition. I tried the  approach below, but didn't get it to work.

```typst
show link.where(body: body not in acronyms.keys()): it => text(fill: blue, it)
```

I essentially need to check if the link contains either a key or a value from the acronym dictionary, as I don't know if the display acronym is the long or short form. How would I do that?

**Comment by Coekjan** [comment, marked as answer]
You could try this:

```typ
#let acronyms = (
  messenger: "MErcury Surface, Space ENvironment, GEochemistry and Ranging",
  lisa: "Laser Interferometer Space Antenna",
)
#show link: it => if it.body.text not in acronyms.keys() { text(fill: blue, it) } else { it }

- #link("messenger")
- #link("lisa")
- #link("this is not acronyms")
```

![image](https://github.com/typst/typst/assets/69834864/83739363-1620-4a8c-8ef4-61ca2cb8e3b2)

**Comment by DannySeidel** [comment]
Thanks with little modifications it solved my problem 🙏🏻
